### PR TITLE
Updates Sonar to 2.5, adds Jacoco and Karma-coverage, fixes 1788

### PR DESCRIPTION
**Pull request by bescott1**
Updates sonar to 2.5.

Adds Jacoco for Java code coverage, and Karma-coverage for Javascript code coverage. Tests results are consolidated in target/test-results/coverage/

The sonar properties are set to grab the coverage reports, and the tests results. I've ignored bower-components sub directories from analysis.

### CODE REVIEW

**Comment by gmarziou** [comment]
Great, you may want to also exclude src/main/webapp/dist

**Comment by atomfrede** [comment]
Any specific reason why you're not using the latest ('0.7.5.201505241946') jacoco version?

File: app/templates/_package.json [hunk 1]
```diff
@@ -62,6 +62,8 @@
     "karma-phantomjs-launcher": "0.2.0",
     "phantomjs": "1.9.17",
     "karma": "0.12.35",
+    "karma-coverage": "^0.4.2",
```
Comment (jdubois) [inline app/templates/_package.json]: Please use fixed versions, like "0.4.2", we've had too many issues with this (I know, I need to do a "policies" document for all this)

**Comment by bescott1** [comment]
@gmarziou I will exclude that sub directory as well

@atomfrede I used the version that I've tested extensively while managing Tatami. At the time it was the latest, I didn't check to see if there was an update. I can use the latest.

@jdubois I will make those changes

**Pull request closed without merging**